models/contacts.js

```diff
@@ -1,19 +1,63 @@
-// const fs = require('fs/promises')
+const fs = require("fs/promises");
+const path = require("path");
+const uuid = require("uuid");
 
-const listContacts = async () => {}
+const contactsPath = path.resolve("./models/contacts.json");
 
-const getContactById = async (contactId) => {}
+const listContacts = async () => {
+  const data = await fs.readFile(contactsPath);
+  const contacts = JSON.parse(data);
+  return contacts;
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
+  const allContacts = await listContacts();
+  const contact = allContacts.find((item) => item.id === contactId);
+  return contact;
+};
 
-const addContact = async (body) => {}
+const removeContact = async (contactId) => {
+  const allContacts = await listContacts();
+  const contactIndex = allContacts.findIndex((item) => item.id === contactId);
+  const deletedContact = allContacts[contactIndex];
+  if (contactIndex !== -1) {
+    allContacts.splice(contactIndex, 1);
+    await fs.writeFile(contactsPath, JSON.stringify(allContacts));
+  }
+  return deletedContact;
+};
 
-const updateContact = async (contactId, body) => {}
+const addContact = async ({ name, email, phone }) => {
+  const newContact = {
+    id: uuid.v4(),
+    name,
+    email,
+    phone,
+  };
+  const allContacts = await listContacts();
+  allContacts.push(newContact);
+  await fs.writeFile(contactsPath, JSON.stringify(allContacts));
+  return newContact;
+};
+
+const updateContact = async (contactId, { name, email, phone }) => {
+  const allContacts = await listContacts();
+  const contactIndex = allContacts.findIndex((item) => item.id === contactId);
+  if (contactIndex !== -1) {
+    allContacts[contactIndex].name = name;
+    allContacts[contactIndex].email = email;
+    allContacts[contactIndex].phone = phone;
+    await fs.writeFile(contactsPath, JSON.stringify(allContacts));
```

Comment:
```suggestion
    await fs.writeFile(contactsPath, JSON.stringify(allContacts, null, 2));
```
додаткові агрументи потрібні для форматування результату. Щоб усе не було в одну строку, як зараз